### ISSUE TITLE
Only 2 channels available on 4 a channel card

### ISSUE THREAD

**Issue by jose1711** [UltraStar-Deluxe/USDX#594]
### Actual behaviour

I only see 2 channels available in Record menu while there are 4 channels input channels provided by the card.

`pavucontrol`:
![obrázok](https://user-images.githubusercontent.com/1406222/141816542-00af6bcb-bdba-4946-afaf-b9efbb457369.png)

capture controls in `alsamixer`:
![obrázok](https://user-images.githubusercontent.com/1406222/141817614-5d51721b-9d8b-48d7-9645-3b56a3445287.png)

### Expected behaviour

There should be 4 channels in the menu so that 4 players can play - each using a separate channel.

### Steps to reproduce

1. Attach [ZEDi10FX](https://www.allen-heath.com/ahproducts/zedi-10fx/) interface/mix via USB and turn it on
2. Compile `ultrastardx` (latest master) and run it
3. Navigate to "Options/Record" menu, select "ZEDi10" soundcard
4. Check number of available channels

### Details

Provide some additional information:

- USDX version: current master (https://github.com/UltraStar-Deluxe/USDX/commit/088aaa2a000baf2f02e69a6eae832097dc4ffe4e)
- Operating System + version: Arch Linux x86_64-rolling, pulseaudio
```
$ pactl list
..
..
Card #85
	Name: alsa_card.usb-Allen___Heath_ZEDi10-00
	Driver: alsa
	Owner Module: n/a
	Properties:
		device.enum.api = "udev"
		device.api = "alsa"
		media.class = "Audio/Device"
		api.alsa.path = "hw:5"
		api.alsa.card = "5"
		api.alsa.card.name = "ZEDi10"
		api.alsa.card.longname = "Allen & Heath ZEDi10 at usb-0000:0c:00.3-4, high speed"
		device.plugged.usec = "848608403"
		device.bus_path = "pci-0000:0c:00.3-usb-0:4:1.0"
		device.sysfs.path = "/sys/devices/pci0000:00/0000:00:08.1/0000:0c:00.3/usb5/5-4/5-4:1.0/sound/card5"
		device.bus-id = "usb-Allen___Heath_ZEDi10-00"
		device.bus = "usb"
		device.subsystem = "sound"
		device.vendor.id = "8944"
		device.vendor.name = "Allen & Heath"
		device.product.id = "22"
		device.product.name = "ZEDi10"
		device.serial = "Allen___Heath_ZEDi10"
		device.name = "alsa_card.usb-Allen___Heath_ZEDi10-00"
		device.description = "ZEDi10"
		device.nick = "ZEDi10"
		device.icon_name = "audio-card-usb"
		api.alsa.use-acp = "true"
		api.acp.auto-profile = "false"
		api.acp.auto-port = "false"
		api.dbus.ReserveDevice1 = "Audio5"
		factory.id = "14"
		client.id = "31"
		object.id = "85"
		object.path = "alsa:pcm:5"
		alsa.card = "5"
		alsa.card_name = "ZEDi10"
		alsa.long_card_name = "Allen & Heath ZEDi10 at usb-0000:0c:00.3-4, high speed"
		alsa.driver_name = "snd_usb_audio"
		device.string = "5"
	Profiles:
		off: Vypnuté (sinks: 0, sources: 0, priority: 0, available: yes)
		output:analog-surround-40+input:analog-surround-40: Analógový priestorový 4.0 výstup + Analógový priestorový 4.0 vstup (sinks: 1, sources: 1, priority: 1212, available: yes)
		output:analog-surround-40: Analógový priestorový 4.0 výstup (sinks: 1, sources: 0, priority: 1200, available: yes)
		input:analog-surround-40: Analógový priestorový 4.0 vstup (sinks: 0, sources: 1, priority: 12, available: yes)
		pro-audio: Pro Audio (sinks: 1, sources: 1, priority: 1, available: yes)
	Active Profile: output:analog-surround-40+input:analog-surround-40
	Ports:
		analog-input-mic: Mikrofón (type: Mic, priority: 8700, latency offset: 0 usec, availability unknown)
			Properties:
				port.type = "mic"
				device.icon_name = "audio-input-microphone"
				card.profile.port = "0"
			Part of profile(s): input:analog-surround-40, output:analog-surround-40+input:analog-surround-40
		analog-output: Analógový výstup (type: Analog, priority: 9900, latency offset: 0 usec, availability unknown)
			Properties:
				port.type = "analog"
				card.profile.port = "1"
			Part of profile(s): output:analog-surround-40, output:analog-surround-40+input:analog-surround-40
```

**Comment by s09bQ5** [comment]
Did you build ultrastardx with PortAudio support? If not, ultrastardx will use SDL and SDL will not allow us to determine the number of channels when used in conjunction with PulseAudio. We tell SDL that we don't care about the number of channels. SDL defaults to 2. PulseAudio complies and mixes all inputs to stereo. You can override SDL's choice with the environment variable SDL_AUDIO_CHANNELS.

**Comment by jose1711** [comment]
Thank you, checked and `PKGBUILD` for Arch indeed contains `--without-portaudio`. I recompiled w/o this option passed all 4 mics are picked up correctly.

**Comment by Tuupertunut** [comment]
This happened to me too with the Behringer UMC404HD usb sound card and 4 microphones. Things I tried:

- Compiling with `--without-portaudio` there were only 2 channels available.
- Compiling with `--with-portaudio` did not work because there is no `portmixer` package available for Ubuntu 24.04 so I could not compile it.
- Compiling without either `--with-portaudio` or `--without-portaudio` worked, but Ultrastar Deluxe was crashing on every other startup to some unhandled alsa error, possibly because the `portmixer` package was still missing.
- Compiling with `--without-portaudio` and setting `SDL_AUDIO_CHANNELS=4` environment variable gave me 4 channels, but audio was still coming only from channels 1 and 2.
- Finally compiling with `--without-portaudio` and setting both `SDL_AUDIO_CHANNELS=4` and `SDL_AUDIODRIVER=pipewire` environment variables made all 4 channels available and without crashing.

This might be useful info for future people trying to run USDX on Ubuntu with 4 microphones.

**Comment by s09bQ5** [comment]
See #753
